### PR TITLE
cherry pick c0ec10d: macOS Keyboard Support (#657)

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

## Summary

Add keyboard events to 0.62-stable

## Changelog

[General] [Added] - Add key events

## Test Plan

Confirmed that this works in RNTester.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-macos/pull/677)

### CODE REVIEW

**Review by harrieshin — APPROVED** [review]

File: Libraries/Components/Touchable/TouchableOpacity.js [hunk 1]
```diff
@@ -164,16 +175,20 @@ class TouchableOpacity extends React.Component<Props, State> {
...
-        if (this.props.onPress != null) {
-          this.props.onPress(event);
+      onKeyUp: event => {
+        if (this.props.onKeyUp != null) {
+          this.props.onKeyUp(event);
         }
       },
+      validKeysDown: this.props.validKeysDown,
+      validKeysUp: this.props.validKeysUp,
+      onLongPress: this.props.onLongPress,
```
Comment (harrieshin) [inline Libraries/Components/Touchable/TouchableOpacity.js:190]: can you keep the existing onlongpress and oppress and just bring the onkeyDown validkeydown stuff?

Comment (amgleitman) [reply Libraries/Components/Touchable/TouchableOpacity.js:190]: Sure, not a bad idea to keep the null checks in place.

File: Libraries/Components/Touchable/TouchableWithoutFeedback.js [hunk 2]
```diff
@@ -253,25 +255,48 @@ class TouchableWithoutFeedback extends React.Component<Props, State> {
 
 function createPressabilityConfig(props: Props): PressabilityConfig {
   return {
-    cancelable: !props.rejectResponderTermination,
```
Comment (harrieshin) [inline Libraries/Components/Touchable/TouchableWithoutFeedback.js]: these props can be deleted?

Comment (amgleitman) [reply Libraries/Components/Touchable/TouchableWithoutFeedback.js]: These props aren't part of `PressabilityConfig` in 0.62-stable. They also make flow sad, so we don't really have much of a choice.

**Pull request merged**